### PR TITLE
Disable flag for TracerProvider / MeterProvider

### LINKED ISSUE #3455: OpenTelemetry .NET Auto-Instrumentation/SDK Side-by-Side Support
# Feature Request

**Is your feature request related to a problem?**

OpenTelemetry .NET Auto-Instrumentation **dynamically loads** OpenTelemetry .NET SDK and its instrumentation to user process and helps collect telemetry from an application without modifying the code. Issue happens when the application has a reference to OpenTelemetry SDK and user want to enable Auto-Instrumentation. In this case both user application and Auto-Instrumentation will have its own instance of `TracerProvider` or `MeterProvider` and could collect duplicate telemetries. 

**Describe the solution you'd like:**

Auto-Instrumentation uses [.NET startup hook](https://github.com/dotnet/runtime/blob/main/docs/design/features/host-startup-hook.md) to load the OpenTelemetry SDK and initialize `TracerProvider`/`MeterProvider` before the main application's entry point. SDK providing an option to disable the creation of the `TracerProvider`/`MeterProvider` for user-initiated providers should solve this issue. This could be implemented by adding a static bool flag in `TracerProvider`/ `MeterProvider` to stop creation of these providers. This flag can be private or public, auto-instrumentation has ability to set this through reflection. If customer use auto-instrumentation, this flag could be set by auto-instrumentation in startup and it will avoid creation of TracerProvider from user application. 

## Additional Context

https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/633

cc: @open-telemetry/dotnet-instrumentation-approvers

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Fixes #3455 .

## Changes

Added disable flag in `MeterProviderBuilderBase` and `TracerProviderBuilderBase`. When disable flag is set, builder will create a default `MeterProvider` or `TracerProvider`. From OpenTelemetry Auto-Instrumentation this could be controlled through reflection.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

@open-telemetry/dotnet-instrumentation-approvers , FYI.

### CODE REVIEW

**Comment by CodeBlanch** [comment]
@rajkumar-rangaraj Would you be open to working on this after #3533 is done? I'm in the middle of changing this stuff not totally sure how this will fit in.

**Comment by rajkumar-rangaraj** [comment]
> @rajkumar-rangaraj Would you be open to working on this after #3533 is done? I'm in the middle of changing this stuff not totally sure how this will fit in.

@CodeBlanch sure, I will wait for #3533 to complete.

**Comment by pellared** [comment]
@rajkumar-rangaraj I see that https://github.com/open-telemetry/opentelemetry-dotnet/pull/3533 is merged. Can you please double-check and clarify if it is ready for review?

File: src/OpenTelemetry/Metrics/MeterProviderBuilderBase.cs [hunk 1]
```diff
@@ -32,6 +32,10 @@ public abstract class MeterProviderBuilderBase : MeterProviderBuilder
     {
         internal const int MaxMetricsDefault = 1000;
         internal const int MaxMetricPointsPerMetricDefault = 2000;
+
+        // Builds noop MeterProvider when this flag is set.
+        // This flag is controlled through reflection.
+        private static bool disableMeterProviderBuilder = false;
```
Comment (reyang) [inline src/OpenTelemetry/Metrics/MeterProviderBuilderBase.cs:38]: Is there a scenario where metrics/traces need to be disabled independently?
What is the contract and how to maintain the contract?

Comment (reyang) [reply src/OpenTelemetry/Metrics/MeterProviderBuilderBase.cs:38]: Maybe consider something similar to https://github.com/open-telemetry/opentelemetry-dotnet/blob/6b7f2dd77cf9d37260a853fcc95f7b77e296065d/src/OpenTelemetry/SuppressInstrumentationScope.cs#L25-L29.

**Comment by rajkumar-rangaraj** [comment]
Will be addressed through #3639

**Pull request closed without merging**